## 32_exim4.dpatch by Andreas Metzler

## DP: The main binary is installed as /usr/sbin/exim4

diff -NurBbp exim-4.71.orig/OS/Makefile-Linux exim-4.71/OS/Makefile-Linux
--- exim-4.71.orig/OS/Makefile-Linux	2009-11-28 10:52:23.000000000 +0100
+++ exim-4.71/OS/Makefile-Linux	2009-11-28 10:53:07.000000000 +0100
@@ -24,9 +24,9 @@ XLFLAGS=-L$(X11)/lib
 X11_LD_LIB=$(X11)/lib
 
 EXIWHAT_PS_ARG=ax
-EXIWHAT_EGREP_ARG='/exim( |$$)'
+EXIWHAT_EGREP_ARG='/exim4( |$$)'
 EXIWHAT_MULTIKILL_CMD=killall
-EXIWHAT_MULTIKILL_ARG=exim
+EXIWHAT_MULTIKILL_ARG=exim4
 EXIWHAT_KILL_SIGNAL=-USR1
 
 # End
diff -NurBbp exim-4.71.orig/src/exicyclog.src exim-4.71/src/exicyclog.src
--- exim-4.71.orig/src/exicyclog.src	2009-11-16 20:50:36.000000000 +0100
+++ exim-4.71/src/exicyclog.src	2009-11-28 10:53:07.000000000 +0100
@@ -145,7 +145,7 @@ done
 
 st='	 '
 exim_path=`grep "^[$st]*exim_path" $config | sed "s/.*=[$st]*//"`
-if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim; fi
+if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim4; fi
 
 spool_directory=`$exim_path -C $config -bP spool_directory | sed 's/.*=[  ]*//'`
 
diff -NurBbp exim-4.71.orig/src/exim_checkaccess.src exim-4.71/src/exim_checkaccess.src
--- exim-4.71.orig/src/exim_checkaccess.src	2009-11-16 20:50:36.000000000 +0100
+++ exim-4.71/src/exim_checkaccess.src	2009-11-28 10:55:19.000000000 +0100
@@ -53,7 +53,7 @@ done
 # a tab to keep the tab in one place.
 
 exim_path=`perl -ne 'chop;if (/^\s*exim_path\s*=\s*(.*)/){print "$1\n";last;}' $config`
-if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim; fi
+if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim4; fi
 
 
 #########################################################################
diff -NurBbp exim-4.71.orig/src/eximon.src exim-4.71/src/eximon.src
--- exim-4.71.orig/src/eximon.src	2004-10-07 12:39:01.000000000 +0200
+++ exim-4.71/src/eximon.src	2009-11-28 10:53:07.000000000 +0100
@@ -66,7 +66,7 @@ config=${EXIMON_EXIM_CONFIG-$config}
 
 st='	 '
 EXIM_PATH=`grep "^[$st]*exim_path" $config | sed "s/.*=[$st]*//"`
-if test "$EXIM_PATH" = ""; then EXIM_PATH=BIN_DIRECTORY/exim; fi
+if test "$EXIM_PATH" = ""; then EXIM_PATH=BIN_DIRECTORY/exim4; fi
 
 SPOOL_DIRECTORY=`$EXIM_PATH -C $config -bP spool_directory | sed 's/.*=[  ]*//'`
 LOG_FILE_PATH=`$EXIM_PATH -C $config -bP log_file_path | sed 's/.*=[  ]*//'`
diff -NurBbp exim-4.71.orig/src/exinext.src exim-4.71/src/exinext.src
--- exim-4.71.orig/src/exinext.src	2009-11-16 20:50:36.000000000 +0100
+++ exim-4.71/src/exinext.src	2009-11-28 10:53:07.000000000 +0100
@@ -91,7 +91,7 @@ if [ "$exim_path" = "" ]; then
   exim_path=`grep "^[$st]*exim_path" $config | sed "s/.*=[$st]*//"`
 fi
 
-if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim; fi
+if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim4; fi
 spool_directory=`$exim_path $eximmacdef -C $config -bP spool_directory | sed 's/.*=[  ]*//'`
 qualify_domain=`$exim_path $eximmacdef -C $config -bP qualify_domain | sed 's/.*=[  ]*//'`
 
@@ -172,7 +172,7 @@ perl - $exim_path "$eximmacdef" $argone 
 
   # Run exim_dumpdb to get out the retry data and pick off what we want
 
-  open(DATA, "${exim}_dumpdb $spool retry |") ||
+  open(DATA, "/usr/sbin/exim_dumpdb $spool retry |") ||
     die "can't run exim_dumpdb";
 
   while (<DATA>)
diff -NurBbp exim-4.71.orig/src/exiqgrep.src exim-4.71/src/exiqgrep.src
--- exim-4.71.orig/src/exiqgrep.src	2004-10-07 12:39:01.000000000 +0200
+++ exim-4.71/src/exiqgrep.src	2009-11-28 10:53:07.000000000 +0100
@@ -22,7 +22,7 @@ use strict;
 use Getopt::Std;
 
 # Have this variable point to your exim binary.
-my $exim = 'BIN_DIRECTORY/exim';
+my $exim = 'BIN_DIRECTORY/exim4';
 my $eargs = '-bpu';
 my %id;
 my %opt;
diff -NurBbp exim-4.71.orig/src/exiwhat.src exim-4.71/src/exiwhat.src
--- exim-4.71.orig/src/exiwhat.src	2009-11-16 20:50:36.000000000 +0100
+++ exim-4.71/src/exiwhat.src	2009-11-28 10:53:07.000000000 +0100
@@ -89,7 +89,7 @@ fi
 
 st='	 '
 exim_path=`grep "^[$st]*exim_path" $config | sed "s/.*=[$st]*//"`
-if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim; fi
+if test "$exim_path" = ""; then exim_path=BIN_DIRECTORY/exim4; fi
 spool_directory=`$exim_path -C $config -bP spool_directory | sed "s/.*=[ ]*//"`
 process_log_path=`$exim_path -C $config -bP process_log_path | sed "s/.*=[ ]*//"`
 
diff -NurBbp exim-4.71.orig/src/globals.c exim-4.71/src/globals.c
--- exim-4.71.orig/src/globals.c	2009-11-16 20:50:37.000000000 +0100
+++ exim-4.71/src/globals.c	2009-11-28 10:53:07.000000000 +0100
@@ -569,7 +569,7 @@ int     errors_sender_rc       = EXIT_FA
 
 gid_t   exim_gid               = EXIM_GID;
 BOOL    exim_gid_set           = TRUE;          /* This gid is always set */
-uschar *exim_path              = US BIN_DIRECTORY "/exim"
+uschar *exim_path              = US BIN_DIRECTORY "/exim4"
                         "\0<---------------Space to patch exim_path->";
 uid_t   exim_uid               = EXIM_UID;
 BOOL    exim_uid_set           = TRUE;          /* This uid is always set */
